modules/kubernetes-addons/crossplane/main.tf

```diff
@@ -17,49 +17,127 @@ module "helm_addon" {
   depends_on = [kubernetes_namespace_v1.crossplane]
 }
 
-resource "kubectl_manifest" "controller_config" {
-  yaml_body = templatefile("${path.module}/aws-provider/controller-config.yaml", {
-    iam-role-arn = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/${var.eks_cluster_id}-provider-aws--irsa"
+#--------------------------------------
+# AWS Provider
+#--------------------------------------
+resource "kubectl_manifest" "aws_controller_config" {
+  count = var.aws_provider.enable == true ? 1 : 0
+
+  yaml_body = templatefile("${path.module}/aws-provider/aws-controller-config.yaml", {
+    iam-role-arn = "arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/${var.eks_cluster_id}-${local.aws_provider_sa}-irsa"
```

Comment:
we have `arn:${data.aws_partition.current.partition}:iam::${data.aws_caller_identity.current.account_id}:role/` in lots of places in this codebase. We should really think about putting this in a variable.

Reply:
I will use local variable to replace this. This way we only make one call to the data resource API. This is my assumption but we can check with hashicorp as well.

Reply:
done